File: frontend/app/lib/shared/data/profiles.ts
import { Profile } from "../../domain/Profile";

export const ALL_PROFILES: readonly Profile[] = [
  new Profile("Profile 1"),
  new Profile("Profile 2"),
  new Profile("Profile 3"),
] as const;

export function getProfileByName(name: string): Profile {
  const result = ALL_PROFILES.find((profile) => profile.name === name);
  if (result === undefined) {
    throw Error();
  }
  return result;
}

File: backend/app/core/data/profiles.py
from typing import List

from app.core.domain import Profile

ALL_PROFILES: List[Profile] = [
    Profile("Profile 1"),
    Profile("Profile 2"),
    Profile("Profile 3"),
]


def get_profile_by_name(name: str) -> Profile:
    profile = None
    for p in ALL_PROFILES:
        if p.name == name:
            profile = p
            break
    if profile is None:
        raise ValueError(f"Profile '{name}' not found")
    return profile
